prac_06/car.py

```diff
@@ -0,0 +1,39 @@
+"""CP1404/CP5632 Practical - Car class."""
+
+
+class Car:
+    """Represent a Car object."""
+
+    def __init__(self, fuel=42, name=""):
+        """Initialise a Car instance.
+
+        fuel: float, one unit of fuel drives one kilometre
+        """
+        self.fuel = fuel
+        self._odometer = 277
+        self.name = name
+
+    def add_fuel(self, amount):
+        """Add amount to the car's fuel."""
+        self.fuel += amount
+
+    def drive(self, distance):
+        """Drive the car a given distance.
+
+        Drive given distance if car has enough fuel
+        or drive until fuel runs out return the distance actually driven.
+        """
+        if distance > self.fuel:
+            distance = self.fuel
+            self.fuel = 0
+        else:
+            self.fuel -= distance
+        self._odometer += distance
+        return distance
+
+    def __str__(self):
+        return f"{self.name}, fuel={self.fuel}, odometer={self._odometer}"
+
+    @property
+    def odometer(self):
+        return self._odometer
```

Comment:
Was this part of the required modifications?

Reply:
I am not sure i don't think so.